### PR TITLE
Retrieve status for bulk requests

### PR DESCRIPTION
**Pull request by alimez**
Added a status command for group_id. By default it only returns the requests and number of successful/failed/running tasks associated with given group_id. Run with --full_report command for full details on each request.

Sample output:
```
python3 turbinia/turbiniactl.py status  -g 2c4393fdb8bb4c7bb6aa5fb1fa1bc1f1

2022-01-25 18:20:26 [INFO] Turbinia version: unknown
2022-01-25 18:20:27 [INFO] Disabling non-allowlisted jobs configured to be disabled in the config file: volatilityjob, dockercontainersenumerationjob, photorecjob


# Turbinia report for group ID 2c4393fdb8bb4c7bb6aa5fb1fa1bc1f1
Request Id a43c7d9fb1ef4d1a90c394737d777cf6 with 1 successful, 6 failed, and 0 running tasks.
Request Id 77952f84a956441c9c1c8235245a0930 with 1 successful, 6 failed, and 0 running tasks.
Request Id 0ca16d5884d3435fbec8bbd4522b3233 with 1 successful, 6 failed, and 0 running tasks.
```

### CODE REVIEW

File: turbinia/client.py [hunk 1]
```diff
@@ -787,8 +790,8 @@ def format_request_status(
 
   def format_task_status(
       self, instance, project, region, days=0, task_id=None, request_id=None,
-      user=None, all_fields=False, full_report=False,
-      priority_filter=Priority.HIGH, output_json=False):
+      group_id=None, user=None, all_fields=False, full_report=False,
+      priority_filter=Priority.HIGH, output_json=False, report=[]):
```
Comment (aarontp) [inline turbinia/client.py:794]: FYI, mutable objects should not be used as default function arguments as they can cause unexpected side effects.   See https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments for details.

File: turbinia/client.py [hunk 2]
```diff
@@ -836,22 +841,52 @@ def format_task_status(
...
     # Build up data
-    report = []
-    requester = task_results[0].get('requester')
-    request_id = task_results[0].get('request_id')
     success_types = ['Successful', 'Failed', 'Scheduled or Running']
     success_values = [True, False, None]
     # Reverse mapping values to types
     success_map = dict(zip(success_values, success_types))
+    # This is used for group ID status
+    requests = defaultdict(dict)
```
Comment (aarontp) [inline turbinia/client.py:849]: Just curious, is the defaultdict giving us anything over a normal dict in this case?

**Review by aarontp — COMMENTED** [review]
LG, just a couple minor things.

File: turbinia/client.py (hunk 2 above)
Comment (alimez) [reply turbinia/client.py:849]: This makes it easier to define a two dimensional dict.

File: turbinia/client.py [hunk 3]
```diff
@@ -836,22 +841,52 @@ def format_task_status(
...
+                request_id, success_counts['Successful'],
+                success_counts['Failed'],
+                success_counts['Scheduled or Running']))
+        if full_report:
+          report.append(
+              self.format_task_status(
+                  instance, project, region, days=0, task_id=None,
+                  request_id=request_id, user=user, all_fields=all_fields,
+                  full_report=full_report, priority_filter=priority_filter,
+                  output_json=output_json, report=report))
```
[the remark this answers is not in the document]
Comment (alimez) [reply turbinia/client.py]: Fixed

**Review by aarontp — APPROVED** [review]
LGTM

**Pull request merged**